x-pack/plugins/lens/public/drag_drop/drag_drop.tsx

```diff
@@ -175,350 +336,383 @@ const DragDropInner = React.memo(function DragDropInner(
     isNotDroppable,
     dragType = 'copy',
     dropType = 'add',
-    dropTo,
-    itemsInGroup,
+    keyboardMode,
+    setKeyboardMode,
+    activeDropTarget,
+    setActiveDropTarget,
+    getAdditionalClassesOnEnter,
   } = props;
 
+  const activeDropTargetMatches =
+    activeDropTarget?.activeDropTarget && activeDropTarget.activeDropTarget.id === value.id;
+
   const isMoveDragging = isDragging && dragType === 'move';
 
   const classes = classNames(
     'lnsDragDrop',
     {
       'lnsDragDrop-isDraggable': draggable,
       'lnsDragDrop-isDragging': isDragging,
-      'lnsDragDrop-isHidden': isMoveDragging,
+      'lnsDragDrop-isHidden': isMoveDragging && !keyboardMode,
       'lnsDragDrop-isDroppable': !draggable,
       'lnsDragDrop-isDropTarget': droppable && dragType !== 'reorder',
-      'lnsDragDrop-isActiveDropTarget': droppable && state.isActive && dragType !== 'reorder',
+      'lnsDragDrop-isActiveDropTarget':
+        droppable && activeDropTargetMatches && dragType !== 'reorder',
       'lnsDragDrop-isNotDroppable': !isMoveDragging && isNotDroppable,
-      'lnsDragDrop-isReplacing': droppable && state.isActive && dropType === 'replace',
+      'lnsDragDrop-isReplacing': droppable && activeDropTargetMatches && dropType === 'replace',
     },
-    state.dragEnterClassNames
-  );
-
-  const dragStart = (e: DroppableEvent) => {
-    // Setting stopPropgagation causes Chrome failures, so
-    // we are manually checking if we've already handled this
-    // in a nested child, and doing nothing if so...
-    if (e.dataTransfer.getData('text')) {
-      return;
+    getAdditionalClassesOnEnter && {
+      [getAdditionalClassesOnEnter()]: activeDropTargetMatches,
     }
-
-    // We only can reach the dragStart method if the element is draggable,
-    // so we know we have DraggableProps if we reach this code.
-    e.dataTransfer.setData('text', (props as DraggableProps).label);
-
-    // Chrome causes issues if you try to render from within a
-    // dragStart event, so we drop a setTimeout to avoid that.
-    setState({ ...state });
-    setTimeout(() => setDragging(value));
-  };
-
-  const dragEnd = (e: DroppableEvent) => {
-    e.stopPropagation();
-    setDragging(undefined);
-  };
+  );
 
   const dragOver = (e: DroppableEvent) => {
     if (!droppable) {
       return;
     }
-
     e.preventDefault();
 
     // An optimization to prevent a bunch of React churn.
-    if (!state.isActive) {
-      setState({
-        ...state,
-        isActive: true,
-        dragEnterClassNames: props.getAdditionalClassesOnEnter
-          ? props.getAdditionalClassesOnEnter()
-          : '',
-      });
+    // todo: replace with custom function ?
+    if (!activeDropTargetMatches) {
+      setActiveDropTarget(value);
```

Comment:
🤔  hmmm, with this condition it's being triggered once when the item is being dragged over. Otherwise, it will be running `setActiveDropTarget` infinitely.